[BUG] before_task_publish_handler missing close_old_connections(), so one dropped DB connection breaks task tracking for hours

### Description

`before_task_publish_handler` in `src/documents/signals/handlers.py` creates a `PaperlessTask` row without first calling `close_old_connections()`. Every other celery signal handler in that file does call it:

| handler | `close_old_connections()` |
| --- | --- |
| `task_prerun_handler` | line 1168 |
| `task_postrun_handler` | line 1199 |
| `task_failure_handler` | line 1260 |
| **`before_task_publish_handler`** (line 1115) | **absent** |

`before_task_publish` fires in whichever process publishes the task. For periodic tasks that is celery beat, which is long-lived and has no request or task boundary, so nothing on this path ever resets a stale connection. Once the database drops the connection (DB restart, network blip, DNS failure) beat keeps reusing the dead one, and every subsequent periodic publish fails.

The task itself still publishes and runs fine. Only the `PaperlessTask` row is lost, so those runs go missing from the Tasks view.

Measured on one instance over 72 hours: 3 actual server-side drops produced 234 `the connection is closed` errors, in 9 discrete episodes lasting between 6 minutes and 14.8 hours. An episode ends when something else in the process happens to reset the connection, which is why the duration varies so much. Restarting the container clears it, but only until the next drop.

Still present on `dev` at the same place, so this is not specific to the beta image.

Suggested fix: call `close_old_connections()` at the top of the `try:` block in `before_task_publish_handler`, matching the sibling handlers.

Searched open and closed issues before filing. #2238, #2588 and #1783 are in the same family but all closed and none covers this handler.

### Steps to reproduce

1. Run paperless-ngx with a periodic task enabled. Mail checking on a 10 minute schedule is enough.
2. Force the database to drop its backends, for example by restarting the postgres container.
3. Watch the beat log.

Expected: beat reconnects and the `PaperlessTask` row is created.

Actual: every subsequent publish logs `Creating PaperlessTask failed` until something else in the process resets the connection.

### Webserver logs

```
[2026-07-28 07:20:00,001] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
[2026-07-28 07:20:00,002] [ERROR] [paperless.handlers] Creating PaperlessTask failed
Traceback (most recent call last):
  File ".../django/db/backends/base/base.py", line 289, in cursor
    return self._cursor()
  File ".../django/db/backends/base/base.py", line 265, in _cursor
    return self._prepare_cursor(self.create_cursor(name))
  File ".../django/db/backends/postgresql/base.py", line 429, in create_cursor
    cursor = self.connection.cursor()
  File ".../psycopg/_connection_base.py", line 532, in _check_connection_ok
    raise e.OperationalError("the connection is closed")
psycopg.OperationalError: the connection is closed

The above exception was the direct cause of the following exception:

django.db.utils.OperationalError: the connection is closed

[2026-07-28 07:20:00,083] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[...] received
[2026-07-28 07:20:00,620] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[...] succeeded in 0.515s: 'No new documents were added.'
```

### Services logs

Postgres, showing the drop that starts an episode:

```
02:01:56 UTC [1] LOG:  untracked child process (PID 22237) exited with exit code 2
02:01:56 UTC [1] LOG:  terminating any other active server processes
02:01:58 UTC [1] LOG:  all server processes terminated; reinitializing
02:02:04 UTC [1] LOG:  database system was not properly shut down; automatic recovery in progress
02:02:04 UTC [1] LOG:  database system is ready to accept connections
```

The first paperless error follows at 02:05:00 UTC and then repeats every 10 minutes. The cause of that postmaster restart is a local problem on my side and is not what this report is about — it is only the trigger that exposes the missing reset.

### Paperless-ngx version

3.0.0

### Host OS

Debian GNU/Linux 12 (bookworm), x86_64

### Installation method

Official Docker image (`ghcr.io/paperless-ngx/paperless-ngx:beta`)

### Configuration changes

None relevant.

--- a/src/documents/signals/handlers.py
+++ b/src/documents/signals/handlers.py
@@ -1134,6 +1134,11 @@ def before_task_publish_handler(
         return
 
     try:
+        # Close stale connections without disrupting a transaction publishing a task
+        for connection in connections.all(initialized_only=True):
+            if not connection.in_atomic_block:
+                connection.close_if_unusable_or_obsolete()
+
         _, task_kwargs, _ = body
         task_id = headers["id"]
 

--- a/src/documents/tests/test_task_signals.py
+++ b/src/documents/tests/test_task_signals.py
@@ -56,6 +56,32 @@ def send_publish(
 
 @pytest.mark.django_db
 class TestBeforeTaskPublishHandler:
+    @mock.patch("documents.signals.handlers.connections.all")
+    def test_closes_old_connections_outside_atomic_blocks(
+        self,
+        connections_all,
+    ) -> None:
+        connection = mock.Mock(in_atomic_block=False)
+        connections_all.return_value = [connection]
+
+        task_id = send_publish("documents.tasks.train_classifier", (), {})
+
+        connection.close_if_unusable_or_obsolete.assert_called_once_with()
+        assert PaperlessTask.objects.filter(task_id=task_id).exists()
+
+    @mock.patch("documents.signals.handlers.connections.all")
+    def test_keeps_connections_open_inside_atomic_blocks(
+        self,
+        connections_all,
+    ) -> None:
+        connection = mock.Mock(in_atomic_block=True)
+        connections_all.return_value = [connection]
+
+        task_id = send_publish("documents.tasks.train_classifier", (), {})
+
+        connection.close_if_unusable_or_obsolete.assert_not_called()
+        assert PaperlessTask.objects.filter(task_id=task_id).exists()
+
     def test_creates_task_for_consume_file(
         self,
         consume_input_doc,

